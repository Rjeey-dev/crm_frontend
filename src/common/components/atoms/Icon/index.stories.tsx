import React from 'react';
import { storiesOf } from '@storybook/react';
import Icon from './index';

storiesOf('Icon', module)
    .add('notifications', () => (
        <Icon classes='mdi mdi-file-document-box'/>
    ))
    .add('settings', () => (
        <Icon classes='mdi mdi-tune'/>
    ));