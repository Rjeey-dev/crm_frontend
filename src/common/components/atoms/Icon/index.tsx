import React from 'react';

interface IProps {
    classes: string,
    attributes?: object
}

function Icon(props: IProps) {
    return <i className={props.classes} {...props.attributes}/>;
}

export default Icon;
