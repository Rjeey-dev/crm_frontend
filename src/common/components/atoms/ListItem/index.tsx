import React, {FormEvent, ReactNode} from "react";

interface IProps {
    children: ReactNode,
    classes?: string,
    handleClick?: (event: FormEvent<HTMLLIElement>) => void,
    attributes?: any
}

function ListItem(props: IProps) {
    return <li className={props.classes} onClick={props.handleClick} {...props.attributes}>{props.children}</li>
}

export default ListItem;