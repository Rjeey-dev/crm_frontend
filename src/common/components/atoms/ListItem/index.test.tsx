import { shallow } from 'enzyme';
import React from 'react';

import Icon from '../Icon/index';
import ListItem from './index';

it("Renders <ListItem/> with correct properties", () => {
    const output = shallow(
        <ListItem classes='class'>
            <Icon classes='icon'/>
        </ListItem>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <ListItem/> with only required props", () => {
    const output = shallow(
        <ListItem>
            <Icon classes='icon'/>
        </ListItem>
    ).html();

    expect(output).toMatchSnapshot();
});