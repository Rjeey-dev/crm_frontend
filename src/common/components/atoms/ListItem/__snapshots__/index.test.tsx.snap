// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ListItem/> with correct properties 1`] = `"<li class=\\"class\\"><i class=\\"icon\\"></i></li>"`;

exports[`Renders <ListItem/> with only required props 1`] = `"<li><i class=\\"icon\\"></i></li>"`;
